pkgs/by-name/ko/koboldcpp/package.nix

```diff
@@ -106,14 +117,28 @@ effectiveStdenv.mkDerivation (finalAttrs: {
   env.NIX_LDFLAGS = lib.concatStringsSep " " (finalAttrs.darwinLdFlags ++ finalAttrs.metalLdFlags);
 
   env.NIX_CFLAGS_COMPILE =
-    lib.optionalString (march != "") "-march=${march}" + lib.optionalString (mtune != "") "-mtune=${mtune}";
+    let
+      isx86_64 = stdenv.hostPlatform.isx86_64;
+      isAarch64 = stdenv.hostPlatform.isAarch64;
+      isDarwin = stdenv.hostPlatform.isDarwin;
+      isLinux = stdenv.hostPlatform.isLinux;
+    in
+    lib.concatStringsSep " " [
+      (lib.optionalString (isx86_64) "-march=x86-64")
+      (lib.optionalString (isAarch64 && isLinux) "-march=armv8.7-a")
+      (lib.optionalString (mtune != "") "-mtune=${mtune}")
+      (lib.optionalString (isDarwin) "-mcpu=native")
```

Comment:
H'm. We try to avoid `-march=native` and similar flags in Nixpkgs, because then the output isn't determined by the derivation structure (="random", "non-deterministic"); instead, we try to describe the host platform statically (=ideally, produces the same output as it would with `-march=native`, but also it'd produce that even if built on a different machine)

Reply:
I was basing that line on: https://stackoverflow.com/a/72597669 *(Forgot to add a comment about this)*

But there is also: https://stackoverflow.com/a/69926065 so I would go with that instead

Reply:
Yes looks much better; regarding the `hostPlatform` and the `cc-wrapper`, I'm seeing this:

```console
nix-repl> :lf nixpkgs
Added 15 variables.
nix-repl> legacyPackages.aarch64-darwin.hostPlatform.gcc.cpu
"apple-a13"
```

and this:

https://github.com/NixOS/nixpkgs/blob/6c12901eb4ccb5befc24543a110dc0865d72b62b/pkgs/build-support/cc-wrapper/default.nix#L253-L254

Reply:
Hmmm that's seems weird the `apple-a13` is the iPhone 11 Series CPU...

Reply:
I would guess that people in https://matrix.to/#/%23macos%3Anixos.org?via=nixos.org&via=matrix.org&via=nixos.dev must know why